lib/commands/audit.js

```diff
@@ -1,395 +1,9 @@
-      return
-    }
-
-    for (const omitType of this.npm.config.get('omit')) {
-      if (node[omitType]) {
-        return
-      }
-    }
-
-    return {
-      name: spec.name,
-      version,
-      type,
-      location: node.location,
-      registry: this.getSpecRegistry(spec),
-    }
-  }
-
-  async verifySignatures (name, version, registry) {
-    const {
-      _integrity: integrity,
-      _signatures,
-      _attestations,
-      _resolved: resolved,
-    } = await pacote.manifest(`${name}@${version}`, {
-      verifySignatures: true,
-      verifyAttestations: true,
-      ...this.buildRegistryConfig(registry),
-      ...this.npm.flatOptions,
-    })
-    const signatures = _signatures || []
-    const result = {
-      integrity,
-      signatures,
-      attestations: _attestations,
-      resolved,
-    }
-    return result
-  }
-
-  async getVerifiedInfo (edge) {
-    const info = this.getValidPackageInfo(edge)
-    if (!info) {
-      return
-    }
-    const { name, version, location, registry, type } = info
-    if (this.checkedPackages.has(location)) {
-      // we already did or are doing this one
-      return
-    }
-    this.checkedPackages.add(location)
-
-    // We only "audit" or verify the signature, or the presence of it, on
-    // packages whose registry returns signing keys
-    const keys = this.keys.get(registry) || []
-    if (keys.length) {
-      this.auditedWithKeysCount += 1
-    }
-
-    try {
-      const { integrity, signatures, attestations, resolved } = await this.verifySignatures(
-        name, version, registry
-      )
-
-      // Currently we only care about missing signatures on registries that provide a public key
-      // We could make this configurable in the future with a strict/paranoid mode
-      if (signatures.length) {
-        this.verifiedSignatureCount += 1
-      } else if (keys.length) {
-        this.missing.push({
-          integrity,
-          location,
-          name,
-          registry,
-          resolved,
-          version,
-        })
-      }
-
-      // Track verified attestations separately to registry signatures, as all
-      // packages on registries with signing keys are expected to have registry
-      // signatures, but not all packages have provenance and publish attestations.
-      if (attestations) {
-        this.verifiedAttestationCount += 1
-      }
-    } catch (e) {
-      if (e.code === 'EINTEGRITYSIGNATURE' || e.code === 'EATTESTATIONVERIFY') {
-        this.invalid.push({
-          code: e.code,
-          message: e.message,
-          integrity: e.integrity,
-          keyid: e.keyid,
-          location,
-          name,
-          registry,
-          resolved: e.resolved,
-          signature: e.signature,
-          predicateType: e.predicateType,
-          type,
-          version,
-        })
-      } else {
-        throw e
-      }
-    }
-  }
-}
+const VerifySignatures = require('../utils/verify-signatures.js')
```

Comment:
Historically these single-use abstractions were a code smell.  One of the reasons for this was because they were usually done from a place of trying to test them in isolation.  The other is cognitive load.  Does making this its own file improve things in a meaningful way here?

Reply:
For me this change reduces cognitive load. `audit` does other things besides verifying signatures, so reducing the file size and removing `require` calls that are only used for verifying signatures made sense to me.